articles/step-by-step-on-how-to-user-nodejs-repl-tutorial/index.md

```diff
@@ -0,0 +1,203 @@
+This symbol acts as an indicator that JavaScript is ready to read and evaluate your code.  
+
+We can test the functionality by printing 'hello world' using REPL as shown below:
+
+```bash
+$ node
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'  // output
+> 
+
+```
+In the script, we declare a variable `hello` and assign the text 'hello world'.  
+Next, we call this variable and press enter, to print the 'hello world' string on the screen. That's it.  
+
+Now how do we exit REPL?
+To exit this tool, press `ctrl + Z on your keyboard.  
+
+Output:
+```bash
+> let hello = 'hello world';
+undefined
+> hello
+'hello world'
+> 
+(To exit, press Ctrl+C again or Ctrl+D or type .exit)
+> 
+
+```
+As shown in the output, you can as well use `ctrl+D` or type `.exit`.  
+
+### Executing code using REPL.
+
+As discussed previously, this tool simplifies work by providing a quick way to test Node.js code without creating files.  
+> It's also important to note that any valid JavaScript code can be executed using REPL.
+
+ 1. Performing Arithmetical operations in REPL
+In the previous example, we printed 'hello world' string on the screen, in this section let's dive in and look at arithmetics:  
+
+In your terminal, start REPL: 
+
+```bash
+$ node
+> 
+
+```
+Let's perform basic addition, subtraction,modulus, division and multipliation respectively:
+Output:  
+
+```bash
+> 2+2
+4
+> 5-2
+3
+> 10%3
+1
+> 20/2
+10
+> 10*2
+20
+> 
+
+```
+2. Performing operations using Node's Math library
+Math library is an object with defined methods and properties. We can therefore use these features to perform some arithmetics.  
+
+Example:
+
+```bash
+---skipped lines
+
+> function getRandomNumber(maximum){
+... return Math.floor(Math.random()*maximum);
+... }
+undefined
+
+```
+
+Output:
+```bash
+> console.log(getRandomNumber(20));
+12
+undefined
+> 
+```
+In REPL, we define `getRandomNumber(args)` and pass it a maximum number. The method uses `Math.floor()` to return a random largest integer less than or equal to a given number.
+
+> It's important to note that `Math.random()` does not provide secure random numbers. Do not use them on security-related tasks.
+
+
+3. Calling Methods (Functions) Using REPL
+
+Normally, we write functions to handle specific tasks. REPL provides an easy way to handle these methods.   
+In JavaScript, we commonly use the `console.log()` global method to print messages. 
+
+Let's enter the following in the prompt: 
+
+```bash
+$ node
+> console.log('Hello world');
+```
+Output: 
+
+```bash 
+Hello world
+undefined // this is the return value of this method, console.log()
+> 
+```
+
+When you press ENTER, the output is displayed as shown above.  
+
+Next, let's create a function to add two numbers:  
+
+```bash
+--- skipped lines
+
+> function addTwoNumbers(firstNumber,lastNumber){
+... console.log(firstNumber+ lastNumber)
+... }
+undefined
+
+```
```

Comment:
Why do we use `...`? And, why do we get `undefined`?

Reply:
Node.js??????